docs/content/using-npm/scripts.md

```diff
@@ -74,51 +75,108 @@ The advantage of doing these things at `prepublish` time is that they can be don
 
 ### Life Cycle Operation Order
 
-#### [`npm publish`](/commands/npm-publish)
+#### [`npm cache add`](/commands/npm-cache)
 
-* `prepublishOnly`
 * `prepare`
+
+#### [`npm ci`](/commands/npm-ci)
+
+* `preinstall`
+* `install`
+* `postinstall`
 * `prepublish`
-* `publish`
-* `postpublish`
+* `preprepare`
+* `prepare`
+* `postprepare`
+
+#### [`npm diff`](/commands/npm-diff)
+
+* `prepare`
+
+#### [`npm env`](/commands/npm-env)
+
+* `env` (You can override the default behavior of `npm env` by defining
+    a custom `env` entry in your `scripts` object)
+
+#### [`npm install`](/commands/npm-install)
+
+These also run when you run `npm install -g <pkg-name>`
+
+* `preinstall`
+* `install`
+* `postinstall`
+* `prepublish`
+* `preprepare`
+* `prepare`
+* `postprepare`
+
+If there is a `binding.gyp` file in the root of your package and you
+haven't defined your own `install` or `preinstall` scripts, npm will
+default the `install` command to compile using node-gyp via `node-gyp
+rebuild`
+
+These are run from the scripts of `<pkg-name>`
 
 #### [`npm pack`](/commands/npm-pack)
 
 * `prepack`
+* `prepare`
 * `postpack`
 
-#### [`npm install`](/commands/npm-install)
+#### [`npm publish`](/commands/npm-publish)
+
+* `prepublishOnly`
+* `prepack`
+* `prepare` (not during `--dry-run`)
+* `postpack`
+* `publish`
+* `postpublish`
+
+#### [`npm rebuild`](/commands/npm-rebuild)
 
 * `preinstall`
 * `install`
 * `postinstall`
+* `prepare` (links only)
```

Comment:
Thanks for the updated documentation... Random person from the community here with a question:

What does "links only" mean here?

Reply:
This is still a WIP PR.  This will be fleshed out more as I continue to rewrite this file.

This means symlinks. `npm link` does use symlinks under the hood, but this also ends up happening for instance when you've installed a package globally.